app/Http/Controllers/Licenses/LicenseCheckinController.php

```diff
@@ -112,4 +112,37 @@ public function store(Request $request, $seatId = null, $backTo = null)
         // Redirect to the license page with error
         return redirect()->route('licenses.index')->with('error', trans('admin/licenses/message.checkin.error'));
     }
+
+    /**
+     * Bulk checkin all license seats
+     *
+     * @author [A. Gianotto] [<snipe@snipe.net>]
+     * @see LicenseCheckinController::create() method that provides the form view
+     * @since [v6.1.1]
+     * @return \Illuminate\Http\RedirectResponse
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+
+    public function bulkCheckin(Request $request, $licenseId) {
+
+        $license = License::findOrFail($licenseId);
+        $this->authorize('checkin', $license);
+
+        $licenseSeats = LicenseSeat::where('license_id', '=', $licenseId)
+            ->whereNotNull('assigned_to')
```

Comment:
Will this handle licenses that are checked out to assets, as well? I think that's `asset_id`